frontend/src/metabase/visualizations/components/LineAreaBarChart.jsx

```diff
@@ -270,20 +271,44 @@ export default class LineAreaBarChart extends Component {
       series,
       settings,
       showTitle,
+      actionButtons,
+      onAddSeries,
+      onEditSeries,
+      onRemoveSeries,
       onChangeCardAndRun,
     } = this.props;
 
     const title = settings["card.title"] || card.name;
     const description = series["card.description"];
-    const data = series._raw || series;
-    const cardIds = new Set(data.map(s => s.card.id));
-    const hasTitle = showTitle && title;
+
+    const rawSeries = series._raw || series;
+    const cardIds = new Set(rawSeries.map(s => s.card.id));
+    const hasTitle = showTitle && settings["card.title"];
+    const hasBreakout = card._breakoutColumn != null;
     const canSelectTitle = cardIds.size === 1 && onChangeCardAndRun;
 
+    const hasMultipleSeries = series.length > 1;
+    const canChangeSeries = onAddSeries || onEditSeries || onRemoveSeries;
+    const hasLegendButtons = !hasTitle && actionButtons;
+    const hasLegend = hasMultipleSeries || canChangeSeries || hasLegendButtons;
+
+    const seriesSettings =
+      settings.series && series.map(single => settings.series(single));
+    const labels = seriesSettings
```

Comment:
nit: it could be slightly more compact
`seriesSettings?.map(...) || series.map(single => single.card.name);`

Reply:
the same with `colors` declatation